src/oemof/tabular/_facade.py

```diff
@@ -168,58 +168,77 @@ def _nominal_value(self):
                     "to_from": self.to_from_capacity,
                 }
             else:
-                return self.capacity
+                if value:
+                    return value
+                else:
+                    return self.capacity
 
-    def _investment(self):
+    def _investment(self, bev=False):
         if not self.expandable:
             self.investment = None
             return self.investment
-        if self.capacity_cost is None:
-            msg = (
-                "If you set `expandable`to True you need to set "
-                "attribute `capacity_cost` of component {}!"
+
+        if bev:
+            self.investment = Investment(
+                ep_costs=0,
+                # ep_costs=self.bev_capacity_cost,
+                maximum=self._get_maximum_additional_invest(
+                    "storage_capacity_potential", "storage_capacity"
+                ),
+                minimum=getattr(self, "minimum_storage_capacity", 0),
+                existing=getattr(self, "storage_capacity", 0),
+                lifetime=getattr(self, "lifetime", None),
+                age=getattr(self, "age", 0),
+                fixed_costs=getattr(self, "fixed_costs", None),
             )
-            raise ValueError(msg.format(self.label))
-        # If storage component
-        if isinstance(self, GenericStorage):
-            # If invest costs/MWH are given
-            if self.storage_capacity_cost is not None:
-                self.investment = Investment(
-                    ep_costs=self.storage_capacity_cost,
-                    maximum=self._get_maximum_additional_invest(
-                        "storage_capacity_potential", "storage_capacity"
-                    ),
-                    minimum=getattr(self, "minimum_storage_capacity", 0),
-                    existing=getattr(self, "storage_capacity", 0),
-                    lifetime=getattr(self, "lifetime", None),
-                    age=getattr(self, "age", 0),
-                    fixed_costs=getattr(self, "fixed_costs", None),
+        else:
```

Comment:
I would not add a new, very specific attribute to investment function.
Instead, you should overwrite `_investment` function in BEV class, in order to simply return the investment you create here.
IMO, we shouldn't change `Facade` class until we really need to.

Reply:
I agree and will move this into a bev classmethod.